native/src/main/java/io/ballerina/lib/ibm.ibmmq/CommonUtils.java

```diff
@@ -18,23 +18,189 @@
+            MQPropertyDescriptor propertyDescriptor = new MQPropertyDescriptor();
+            Object propertyObject = mqMessage.getObjectProperty(propertyName, propertyDescriptor);
+            if (propertyObject instanceof Integer intProperty) {
+                property.put(PROPERTY_VALUE, intProperty.longValue());
+            } else if (propertyObject instanceof String stringProperty) {
+                property.put(PROPERTY_VALUE, StringUtils.fromString(stringProperty));
+            } else {
+                property.put(PROPERTY_VALUE, propertyObject);
+            }
+            property.put(PROPERTY_DESCRIPTOR,
+                    populateDescriptorFromMQPropertyDescriptor(propertyDescriptor));
+            properties.put(StringUtils.fromString(propertyName), property);
+        }
+        return properties;
+    }
+
+    private static void populateMQProperties(BMap<BString, Object> properties, MQMessage mqMessage) {
+        for (BString key : properties.getKeys()) {
+            try {
+                handlePropertyValue(properties, mqMessage, key);
+            } catch (MQException e) {
+                throw createError(IBMMQ_ERROR,
+                        String.format("Error occurred while setting message properties: %s", e.getMessage()), e);
+            }
+        }
+    }
+
+    private static void handlePropertyValue(BMap<BString, Object> properties, MQMessage mqMessage, BString key)
+            throws MQException {
+        BMap<BString, Object> property = (BMap<BString, Object>) properties.getMapValue(key);
+        MQPropertyDescriptor propertyDescriptor = defaultPropertyDescriptor;
+        if (property.containsKey(PROPERTY_DESCRIPTOR)) {
+            propertyDescriptor = getMQPropertyDescriptor(properties.getMapValue(PROPERTY_DESCRIPTOR));
+        }
+        Object value = property.get(PROPERTY_VALUE);
+        if (value instanceof Long longValue) {
+            mqMessage.setIntProperty(key.getValue(), propertyDescriptor, longValue.intValue());
+        } else if (value instanceof Boolean booleanValue) {
+            mqMessage.setBooleanProperty(key.getValue(), propertyDescriptor, booleanValue);
+        } else if (value instanceof Byte byteValue) {
+            mqMessage.setByteProperty(key.getValue(), propertyDescriptor, byteValue);
+        } else if (value instanceof byte[] bytesValue) {
+            mqMessage.setBytesProperty(key.getValue(), propertyDescriptor, bytesValue);
+        } else if (value instanceof Float floatValue) {
+            mqMessage.setFloatProperty(key.getValue(), propertyDescriptor, floatValue);
+        } else if (value instanceof Double doubleValue) {
+            mqMessage.setDoubleProperty(key.getValue(), propertyDescriptor, doubleValue);
+        } else if (value instanceof BString stringValue) {
+            mqMessage.setStringProperty(key.getValue(), propertyDescriptor, stringValue.getValue());
+        }
```

Comment:
What should be the default behaviour if the value doesn't fall into any of the above types? Don't we need to have an `else` clause to handle such situations?

Reply:
I think we can only provide the above types from ballerina side as values hence no need to consider `else` scenario. We can just consider the last condition as the `else` case but then we have to manually cast the `value` to a `BString`.